syntax = "proto3";

package inference;

service InferenceService {
    rpc Infer(InferRequest) returns (InferResponse);
    rpc GetMetrics(Empty) returns (MetricsResponse);
}

message Empty {}

message QueryDocPair {
    string query = 1;
    string document = 2;
}

message InferRequest {
    repeated QueryDocPair pairs = 1;
}

message InferResponse {
    repeated float scores = 1;      // Relevance scores for each pair
    int32 num_pairs = 2;            // Number of pairs processed
    float latency_ms = 3;
    int32 status_code = 4;          // HTTP-like status code (200 or 204)
}

message MetricsResponse {
    int32 count = 1;
    float avg_ms = 2;
    float p50_ms = 3;
    float p95_ms = 4;
    float p99_ms = 5;
    float throughput_qps = 6;  // Pairs per second
    int32 query_count = 7;     // Total pairs processed
}
